[Feature] new UI

vedansh-5 opened the pull request:
Closes #67 
stand alone popup UI

## Summary by Sourcery

Redesign the popup UI with a modern layout using Tailwind CSS, add dark mode support, and enhance the scrum report generation and copy functionality. Remove legacy GSoC/CodeHeat logic and refactor scrum data handling for improved flexibility and user experience.

Enhancements:
- Redesign the popup interface with Tailwind CSS for a modern look and improved usability.
- Add dark mode support and toggle functionality to the popup UI.
- Refactor scrum report generation to support both email and popup outputs, including improved error handling and user feedback.
- Remove GSoC/CodeHeat-specific logic in favor of more flexible date range selection (last week, yesterday, or custom).
- Improve scrum report copy functionality to output markdown-formatted text.

Build:
- Add Tailwind CSS, PostCSS, and Autoprefixer to the build system and update related configuration files.

Documentation:
- Update user-facing documentation in the popup UI to reflect new features and usage.

vedansh-5 commented:
This PR has correct fetch logic but lacks some basic and some important tweaks.

vedansh-5 commented:
In these latest changes I have added a toggleRadio function and fixed the enable disable toggle.

vedansh-5 commented:
In the latest changes I fixed the toggleRadio functionality and improved the enable disable toggle.

vedansh-5 commented:
In this commit I have added dark mode and improved visuals.

vedansh-5 commented:
@hongquan I have done a lot in this draft pr, and a lot needs to be done, I'll move onto performance optimization and writing tests, in the meanwhile I'd truly appreciate if you could take a look at this and then maybe we can discuss additional sorting parameters for the data.
Thanks!

vedansh-5 commented:
@mariobehling @hongquan This is the refactored UI, with added scrum standalone, please have a look.

vedansh-5 commented:
@mariobehling @hongquan This UI might have some bugs, I am working on finding and fixing them, but I'd need your approval to know whether I'm moving in the right direction or not.

Preeti9764 commented on src/scripts/popup.js:
```diff
@@ -0,0 +1,139 @@
+        // Toggle icon
+        this.src = isDarkMode ? 'icons/light-mode.png' : 'icons/night-mode.png';
+    });
+})
```
Not sure for this part currently

Preeti9764 reviewed (changes requested):
The overall changes look good.
However, I’m not fully sure about including the scrum preview generation feature at this point. It might be better to hold off on that part for now.

vedansh-5 commented:
> The overall changes look good. However, I’m not fully sure about including the scrum preview generation feature at this point. It might be better to hold off on that part for now.

The scrum part preview is something we need to discuss with @mariobehling @hongquan as decided. This PR is not ready to be merged yet, it has some bugs, also the radio for last one day doesn't work yet, I have not written the logic to it. This PR is intended (as of in its current state) to know whether we're moving in the right direction or not. Thanks for your feedback.

vedansh-5 commented:
In these latest changes I have replaced jQuery's `$.ajax` calls with `async/await` and now using Parallel promises to optimize the data fetching API's speed.
Github API returns a max of 100 results per pages, in this case if we only fetch page 1 we might miss data, so I have added pagination.

hpdang commented:
Also change replace the last question What is stopping you from doing your work with "What is blocking you from making progress?"

hpdang commented:
Please make the following changes: 

Notes: 
- The PRs fetched are based on the most recent review by any contributor. If you reviewed a PR 10 days ago and someone else reviewed it 2 days ago, it will still appear in your activity for the past week. (See [this issue](https://github.com/fossasia/scrum_helper/issues/20))
- Please note that some discrepancies may occur in the generated SCRUM. We recommend manually reviewing and editing the report to ensure accuracy before sharing

@vedansh-5 to confirm: based on note 1, someone reviewed a PR 10 days ago, it will still show in his/her last 7 days activity? 

Also please unbold the top description.

vedansh-5 commented:
> Also please unbold the top description.
sure

vedansh-5 commented:
> @vedansh-5 to confirm: based on note 1, someone reviewed a PR 10 days ago, it will still show in his/her last 7 days activity?

Yes, if the pr has been updated it will show in last 7 days

hpdang commented:
@vedansh-5 When I select the last 1 day, the content/heading questions still show last 7 days 

@Preeti9764 please also help to test this, let's aim to release it today

vedansh-5 commented:
> @vedansh-5 When I select the last 1 day, the content/heading questions still show last 7 days
> 
> @Preeti9764 please also help to test this, let's aim to release it today

Thanks for pointing it out, I am aware of this, I'll fix it and push the code along with working standalone.

Preeti9764 commented:
> @Preeti9764 please also help to test this, let's aim to release it today
okay ,let me check this out.

vedansh-5 commented:
Fixed the radios. Will now work on the standalone scrum generation. In the meantime @Preeti9764 could you take a look #94 for the new rebase errors. Thanks

vedansh-5 commented:
> @vedansh-5 very nice! Could you please also change the blocker question and the Notes part?
> 
> * Change "What is stopping you from doing your work?" to "**What is blocking you from making progress?"** (make it the same for 7 days and 1 day)
> * Change the entire Note section to below:
> 
> Notes:
> 
> * The PRs fetched are based on the most recent review by any contributor. If you reviewed a PR 10 days ago and someone else reviewed it 2 days ago, it will still appear in your activity for the past week. (See [this issue](https://github.com/fossasia/scrum_helper/issues/20))
> * Please note that some discrepancies may occur in the generated SCRUM. We recommend manually reviewing and editing the report to ensure accuracy before sharing

sorry to have missed it, I'll do it right away.

hpdang commented:
> Change "What is stopping you from doing your work?" to "What is blocking you from making progress?"

This should apply in the Email content too. Thank you!

Preeti9764 commented:
> @hpdang This PR is ready to be merged with all the functionalities, please take a look whenever you can. @Preeti9764 Please take a look too, lmk if you encounter any issues. Thanks!

I have tested the new UI and it is all good and ready to be merged . Thanks!

hpdang commented:
> > @hpdang This PR is ready to be merged with all the functionalities, please take a look whenever you can. @Preeti9764 Please take a look too, lmk if you encounter any issues. Thanks!
> 
> I have tested the new UI and it is all good and ready to be merged . Thanks!

@Preeti9764 did you also test in Yahoo and Outlook? Thanks

Preeti9764 commented:
> @Preeti9764 did you also test in Yahoo and Outlook? Thanks

yes,I tested on all the plateforms

Preeti9764 commented:
for google groups, yahoo,outlook respectively ..
![image](https://github.com/user-attachments/assets/8bb7e607-bec6-4a22-a26b-bf47a80900d9)
![Screenshot 2025-06-10 192245](https://github.com/user-attachments/assets/643ebba6-675e-4f0f-a9c7-6d6e26ae299a)
![Screenshot 2025-06-10 192424](https://github.com/user-attachments/assets/2e40aed5-4a53-48e5-a373-f057031470aa)
